### PR TITLE
Module projects rewrite

### PR DESCRIPTION
**Pull request by ncgoodbody**
Rewrote sprint 2 module projects so that 

1. Tasks were more clearly stated for students
2. Tasks aligned with content taught in Guided Projects

### CODE REVIEW

**Review by csslambda — APPROVED** [review]

**Review by csslambda — APPROVED** [review]

File: module1-decision-trees/LS_DS_221_assignments.ipynb [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    "    DATA_PATH = '../data/'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Module Project: Decision Trees\n",
+    "\n",
+    "This week, the module projects will focus on creating and improving a model for the Tanazania Water Pump datset. Your goal is to create a model to predict whether a water pump is functional, non-functional, or needs repair.\n",
```
Comment (nwhoffman) [inline module1-decision-trees/LS_DS_221_assignments.ipynb:40]:
```suggestion
    "This week, the module projects will focus on creating and improving a model for the Tanazania Water Pump dataset. Your goal is to create a model to predict whether a water pump is functional, non-functional, or needs repair.\n",
```

File: module1-decision-trees/LS_DS_221_assignments.ipynb [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    "\n",
+    "Dataset source: [DrivenData.org](https://www.drivendata.org/competitions/7/pump-it-up-data-mining-the-water-table/).\n",
+    "\n",
+    "## Directions\n",
+    "\n",
+    "The task for this project are as follows:\n",
```
Comment (nwhoffman) [inline module1-decision-trees/LS_DS_221_assignments.ipynb:46]:
```suggestion
    "The tasks for this project are as follows:\n",
```

File: module1-decision-trees/LS_DS_221_assignments.ipynb [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+   "outputs": [],
+   "source": [
+    "%%capture\n",
+    "import sys\n",
+    "\n",
+    "# If you're on Colab:\n",
+    "if 'google.colab' in sys.modules:\n",
+    "    !pip install category_encoders==2.*\n",
+    "    !pip install pandas-profiling==2.*\n",
+    "    DATA_PATH = 'https://raw.githubusercontent.com/LambdaSchool/DS-Unit-2-Kaggle-Challenge/master/data/'\n",
```
Comment (nwhoffman) [inline module1-decision-trees/LS_DS_221_assignments.ipynb:27]:
```suggestion
    "    DATA_PATH = 'https://raw.githubusercontent.com/LambdaSchool/DS-Unit-2-Kaggle-Challenge/main/data/'\n",
```

Comment (nwhoffman) [reply module1-decision-trees/LS_DS_221_assignments.ipynb:27]: With the default branch changed to main, I think this might need to be updated?

File: module1-decision-trees/LS_DS_221_assignments.ipynb [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+    "print('Validation Accuracy Score:', ...)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# VI. Tune Model\n",
+    "\n",
+    "**Task 8:** Is there a large differnce between your training and validation accuracy? If so, experiment with different setting for `max_depth` in your `DecisionTreeClassifier` to reduce the amount of overfitting in your model."
```
Comment (nwhoffman) [inline module1-decision-trees/LS_DS_221_assignments.ipynb:251]:
```suggestion
    "**Task 8:** Is there a large difference between your training and validation accuracy? If so, experiment with different setting for `max_depth` in your `DecisionTreeClassifier` to reduce the amount of overfitting in your model."
```

File: module2-random-forests/LS_DS_222_assignment.ipynb [hunk 5]
```diff
@@ -12,171 +12,302 @@
...
+    "Dataset source: [DrivenData.org](https://www.drivendata.org/competitions/7/pump-it-up-data-mining-the-water-table/).\n",
     "\n",
-    "- **\"Categorical Encoding\":** This means using the raw categorical values as-is, not encoded. Scikit-learn doesn't support this, but some tree algorithm implementations do. For example, [Catboost](https://catboost.ai/), or R's [rpart](https://cran.r-project.org/web/packages/rpart/index.html) package.\n",
-    "- **Numeric Encoding:** Synonymous with Label Encoding, or \"Ordinal\" Encoding with random order. We can use [category_encoders.OrdinalEncoder](https://contrib.scikit-learn.org/category_encoders/ordinal.html).\n",
-    "- **One-Hot Encoding:** We can use [category_encoders.OneHotEncoder](https://contrib.scikit-learn.org/category_encoders/onehot.html).\n",
-    "- **Binary Encoding:** We can use [category_encoders.BinaryEncoder](https://contrib.scikit-learn.org/category_encoders/binary.html).\n",
+    "## Directions\n",
     "\n",
+    "The task for this project are as follows:\n",
```
Comment (csslambda) [inline module2-random-forests/LS_DS_222_assignment.ipynb:51]:
```suggestion
    "The tasks for this project are as follows:\n",
```

File: module2-random-forests/LS_DS_222_assignment.ipynb [hunk 6]
```diff
@@ -12,171 +12,302 @@
...
+   "metadata": {},
+   "source": [
+    "# II. Split Data\n",
     "\n",
-    "```python\n",
-    "encoder = ce.TargetEncoder(min_samples_leaf=..., smoothing=...) # Both parameters > 1 to avoid overfitting\n",
-    "X_train_encoded = encoder.fit_transform(X_train, y_train=='functional')\n",
-    "X_val_encoded = encoder.transform(X_train, y_val=='functional')\n",
-    "```\n",
+    "**Task 3:** Split your DataFrame `df` into a feature matrix `X` and the target vector `y`. You want to predict `'pump_status'`."
```
Comment (csslambda) [inline module2-random-forests/LS_DS_222_assignment.ipynb:132]:
```suggestion
    "**Task 3:** Split your DataFrame `df` into a feature matrix `X` and the target vector `y`. You want to predict `'status_group'`."
```

File: module3-cross-validation/LS_DS_223_assignment.ipynb [hunk 7]
```diff
@@ -8,82 +8,259 @@
...
+   "metadata": {},
+   "source": [
+    "# II. Split Data\n",
     "\n",
-    "### Doing\n",
-    "- Add your own stretch goals!\n",
-    "- Try other [categorical encodings](https://contrib.scikit-learn.org/categorical-encoding/). See the previous assignment notebook for details.\n",
-    "- In additon to `RandomizedSearchCV`, scikit-learn has [`GridSearchCV`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html). Another library called scikit-optimize has [`BayesSearchCV`](https://scikit-optimize.github.io/notebooks/sklearn-gridsearchcv-replacement.html). Experiment with these alternatives.\n",
-    "- _[Introduction to Machine Learning with Python](http://shop.oreilly.com/product/0636920030515.do)_ discusses options for \"Grid-Searching Which Model To Use\" in Chapter 6:\n",
+    "**Task 2:** Split your DataFrame `df` into a feature matrix `X` and the target vector `y`. You want to predict `'pump_status'`.\n",
```
Comment (csslambda) [inline module3-cross-validation/LS_DS_223_assignment.ipynb:127]:
```suggestion
    "**Task 2:** Split your DataFrame `df` into a feature matrix `X` and the target vector `y`. You want to predict `'status_group'`.\n",
```

File: module2-random-forests/LS_DS_222_assignment.ipynb [hunk 8]
```diff
@@ -12,171 +12,302 @@
...
-    "test = pd.read_csv(DATA_PATH+'waterpumps/test_features.csv')\n",
-    "sample_submission = pd.read_csv(DATA_PATH+'waterpumps/sample_submission.csv')\n",
+    "assert len(y_pred) == len(X_test), f'Your list of predictions should have {len(X_test)} items in it. '"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Task 11 `stretch goal`:** Create a DataFrame `submission` whose index is the same as `X_test` and that has one column `'pump_status'` with your predictions. Next, save this DataFrame as a CSV file and upload your submissions to our competition site. \n",
```
Comment (csslambda) [inline module2-random-forests/LS_DS_222_assignment.ipynb:298]:
```suggestion
    "**Task 11 `stretch goal`:** Create a DataFrame `submission` whose index is the same as `X_test` and that has one column `'status_group'` with your predictions. Next, save this DataFrame as a CSV file and upload your submissions to our competition site. \n",
```

File: module3-cross-validation/LS_DS_223_assignment.ipynb [hunk 9]
```diff
@@ -92,42 +269,85 @@
...
+   "metadata": {},
+   "source": [
+    "# Communicate Results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Task 9:** Create a DataFrame `submission` whose index is the same as `X_test` and that has one column `'pump_status'` with your predictions. Next, save this DataFrame as a CSV file and upload your submissions to our competition site. \n",
```
Comment (csslambda) [inline module3-cross-validation/LS_DS_223_assignment.ipynb:338]:
```suggestion
    "**Task 9:** Create a DataFrame `submission` whose index is the same as `X_test` and that has one column `'status_group'` with your predictions. Next, save this DataFrame as a CSV file and upload your submissions to our competition site. \n",
```

File: module4-classification-metrics/LS_DS_224_assignment.ipynb [hunk 10]
```diff
@@ -11,82 +11,314 @@
...
+    "X_test = ..."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# II. Split Data\n",
+    "\n",
+    "**Task 2:** Split your DataFrame `df` into a feature matrix `X` and the target vector `y`. You want to predict `'pump_status'`.\n",
```
Comment (csslambda) [inline module4-classification-metrics/LS_DS_224_assignment.ipynb:132]:
```suggestion
    "**Task 2:** Split your DataFrame `df` into a feature matrix `X` and the target vector `y`. You want to predict `'status_group'`.\n",
```

File: module1-decision-trees/LS_DS_221_assignments.ipynb [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+    "X_test = ..."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# II. Split Data\n",
+    "\n",
+    "**Task 3:** Split your DataFrame `df` into a feature matrix `X` and the target vector `y`. You want to predict `'pump_status'`."
```
Comment (nwhoffman) [inline module1-decision-trees/LS_DS_221_assignments.ipynb:134]:
```suggestion
    "**Task 3:** Split your DataFrame `df` into a feature matrix `X` and the target vector `y`. You want to predict `'status_group'`."
```

File: module2-random-forests/LS_DS_222_assignment.ipynb [hunk 12]
```diff
@@ -12,171 +12,302 @@
...
-    "- [Effort Shock and Reward Shock (How The Karate Kid Ruined The Modern World)](http://www.tempobook.com/2014/07/09/effort-shock-and-reward-shock/)\n",
-    "- [How to manage impostor syndrome in data science](https://towardsdatascience.com/how-to-manage-impostor-syndrome-in-data-science-ad814809f068)\n",
-    "- [\"I am not a real data scientist\"](https://brohrer.github.io/imposter_syndrome.html)\n",
-    "- _**[Imposter Syndrome in Data Science](https://caitlinhudon.com/2018/01/19/imposter-syndrome-in-data-science/)**_\n",
+    "%%capture\n",
+    "import sys\n",
     "\n",
+    "# If you're on Colab:\n",
+    "if 'google.colab' in sys.modules:\n",
+    "    DATA_PATH = 'https://raw.githubusercontent.com/LambdaSchool/DS-Unit-2-Linear-Models/master/data/'\n",
```
Comment (nwhoffman) [inline module2-random-forests/LS_DS_222_assignment.ipynb:25]:
```suggestion
    "    DATA_PATH = 'https://raw.githubusercontent.com/LambdaSchool/DS-Unit-2-Linear-Models/main/data/'\n",
```

Comment (nwhoffman) [reply module2-random-forests/LS_DS_222_assignment.ipynb:25]: Change master to main

File: module3-cross-validation/LS_DS_223_assignment.ipynb [hunk 13]
```diff
@@ -8,82 +8,259 @@
...
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Cross-Validation\n",
+    "# Module Project: Hyperparameter Tuning\n",
+    "\n",
+    "This sprint, the module projects will focus on creating and improving a model for the Tanazania Water Pump datset. Your goal is to create a model to predict whether a water pump is functional, non-functional, or needs repair.\n",
```
Comment (nwhoffman) [inline module3-cross-validation/LS_DS_223_assignment.ipynb:42]:
```suggestion
    "This sprint, the module projects will focus on creating and improving a model for the Tanazania Water Pump dataset. Your goal is to create a model to predict whether a water pump is functional, non-functional, or needs repair.\n",
```

File: module3-cross-validation/LS_DS_223_assignment.ipynb [hunk 14]
```diff
@@ -8,82 +8,259 @@
...
-    "Here's some code you can use to \"stack\" multiple submissions, which is another form of ensembling:\n",
+    "**Task 4:** Build a `Pipeline` named `clf_dt`. Your `Pipeline` should include:\n",
     "\n",
-    "```python\n",
-    "import pandas as pd\n",
+    "- an `OrdinalEncoder` transformer for categorical features.\n",
+    "- a `SimpleImputer` transformer fot missing values.\n",
+    "- a `DecisionTreeClassifier` Predictor.\n",
+    "\n",
+    "**Note:** Do not train `clf_dt`. You'll do that in a subsecuent task. "
```
Comment (nwhoffman) [inline module3-cross-validation/LS_DS_223_assignment.ipynb:181]:
```suggestion
    "**Note:** Do not train `clf_dt`. You'll do that in a subsequent task. "
```

File: module3-cross-validation/LS_DS_223_assignment.ipynb [hunk 15]
```diff
@@ -8,82 +8,259 @@
...
-    "files = ['submission-01.csv', 'submission-02.csv', 'submission-03.csv']\n",
+    "- an `OrdinalEncoder` transformer for categorical features.\n",
+    "- a `SimpleImputer` transformer fot missing values.\n",
+    "- a `RandomForestClassifier` predictor.\n",
     "\n",
-    "target = 'status_group'\n",
-    "submissions = (pd.read_csv(file)[[target]] for file in files)\n",
-    "ensemble = pd.concat(submissions, axis='columns')\n",
-    "majority_vote = ensemble.mode(axis='columns')[0]\n",
+    "**Note:** Do not train `clf_rf`. You'll do that in a subsecuent task. "
```
Comment (nwhoffman) [inline module3-cross-validation/LS_DS_223_assignment.ipynb:203]:
```suggestion
    "**Note:** Do not train `clf_rf`. You'll do that in a subsequent task. "
```

File: module4-classification-metrics/LS_DS_224_assignment.ipynb [hunk 16]
```diff
@@ -11,82 +11,314 @@
...
    "cell_type": "markdown",
    "metadata": {
     "colab_type": "text",
     "id": "nCc3XZEyG3XV"
    },
    "source": [
-    "# Classification Metrics\n",
+    "# Module Project: Classification Metrics\n",
+    "\n",
+    "This sprint, the module projects will focus on creating and improving a model for the Tanazania Water Pump datset. Your goal is to create a model to predict whether a water pump is functional, non-functional, or needs repair.\n",
```
Comment (nwhoffman) [inline module4-classification-metrics/LS_DS_224_assignment.ipynb:43]:
```suggestion
    "This sprint, the module projects will focus on creating and improving a model for the Tanazania Water Pump dataset. Your goal is to create a model to predict whether a water pump is functional, non-functional, or needs repair.\n",
```

File: module4-classification-metrics/LS_DS_224_assignment.ipynb [hunk 17]
```diff
@@ -11,82 +11,314 @@
...
-    "- [ ] Plot a confusion matrix for your Tanzania Waterpumps model.\n",
-    "- [ ] Continue to participate in our Kaggle challenge. Every student should have made at least one submission that scores at least 70% accuracy (well above the majority class baseline).\n",
-    "- [ ] Submit your final predictions to our Kaggle competition. Optionally, go to **My Submissions**, and _\"you may select up to 1 submission to be used to count towards your final leaderboard score.\"_\n",
-    "- [ ] Commit your notebook to your fork of the GitHub repo.\n",
-    "- [ ] Read [Maximizing Scarce Maintenance Resources with Data: Applying predictive modeling, precision at k, and clustering to optimize impact](http://archive.is/DelgE), by Lambda DS3 student Michael Brady. His blog post extends the Tanzania Waterpumps scenario, far beyond what's in the lecture notebook.\n",
+    "- **Task 1:** Use `wrangle` function to import training and test data.\n",
+    "- **Task 2:** Split training data into feature matrix `X` and target vector `y`.\n",
+    "- **Task 3:** Split trainig data into training and validation sets.\n",
```
Comment (nwhoffman) [inline module4-classification-metrics/LS_DS_224_assignment.ipynb:53]:
```suggestion
    "- **Task 3:** Split training data into training and validation sets.\n",
```

**Review by nwhoffman — APPROVED** [review]
Review finished! I focused on typos, etc. and not the actual tasks.

**Comment by ncgoodbody** [comment]
Thank you both! I'm going to clean up the commit history and then merge.
